CMakeLists.txt

```diff
@@ -127,10 +134,11 @@ if(DEFINED EMSCRIPTEN)
     add_link_options(-sEXIT_RUNTIME)
 endif()
 
-add_library(${PROJECT_NAME} STATIC ${LIBCORO_SOURCE_FILES})
-set_target_properties(${PROJECT_NAME} PROPERTIES LINKER_LANGUAGE CXX PREFIX "")
+add_library(${PROJECT_NAME} ${LIBCORO_SOURCE_FILES})
```

Comment:
I'm confused how this produces a STATIC lib when the new option is disabled? Maybe I'm just missing something obvious?

Reply:
I see, `BUILD_SHARED_LIBS` is a common `cmake` feature that propogates to all libraries built. Cool.

Reply:
Sorry didn't answering before, I was on vacation. Correct, `BUILD_SHARED_LIBS` is a common feature, it produces static by default. It's a pretty old and backward compatible feature: https://cmake.org/cmake/help/v3.0/variable/BUILD_SHARED_LIBS.html